Add presidio pii detection post (#2187)
huggingface/blog: presidio-pii-detection.md

---
title: "[Presidio] Experimenting with Automatic PII Detection on the Hub" 

> thumbnail: /blog/assets/presidio-pii-detection/presidio_report.jpg

Reviewer: ```suggestion
thumbnail: /blog/assets/presidio-pii-detection/presidio_report.png
```

Reviewer: might not look super good on /blog though, not sure

authors:
- user: lhoestq
---

# [Presidio] Experimenting with Automatic PII Detection on the Hub

> At Hugging Face, we've noticed an interesting trend in machine learning (ML) datasets – the emergence of Personal Identifiable Information (PII), which poses some unique challenges for ML practitioners.

Reviewer 2: ```suggestion
At Hugging Face, working closely with machine learning (ML) datasets hosted on our Hub, we've noticed a concerning trend – the increasing prevalence of undocumented Personal Identifiable Information (PII), which poses some unique challenges for ML practitioners.
```

We're not the only ones to have noticed ;)

Reviewer 2: LGTM otherwise 🤗

> At Hugging Face, working closely with machine learning (ML) datasets hosted on our Hub, we've noticed a concerning trend – the increasing prevalence of undocumented Personal Identifiable Information (PII), which poses some unique challenges for ML practitioners.

Reviewer 3: I added a comment on this riffing off of @yjernite 's comment, here: https://github.com/huggingface/blog/commit/12d18c386c839a0de0ec511cccb7290fcfed2699
But stepping back a bit, note that PII is a term defined by the U.S. military, and similar notions of PII are called other things in other countries, such as ``personal information'' or ``personal data'' (see https://gdpr.eu/eu-gdpr-personal-data/). 
So if this were a post I was penning (just my preference), I'd try to use the term personal information instead, as a generic term that is a bit more internationally sensitive, and then I would later hone in on PII as what the focus is wrt using Presidio.

Reviewer: agree, or at least explain / introduce the acronym clearly

Reviewer 3: Like this: 
"At Hugging Face, we've noticed a concerning trend in machine learning (ML) datasets hosted on our Hub: Undocumented private information about individuals. This poses some unique challenges for ML practitioners."

Author: Sounds good, I went with your suggestion, thanks !

> In this blog post, we'll explore different types of datasets containing PII, the issues they present, and a new feature we're experimenting with on the Dataset Hub to help address these challenges.

Reviewer 3: If you use the term "personal data" above, then here you could make the switch to PII if you wanted to make the least changes wrt using these term.
Something like this:
"In this blog post, we'll explore different types of datasets containing private information known as Personally Identifying Information (PII), the issues they present, ...[etc]"

Author: good idea !

## Types of Datasets with PII

We noticed two types of datasets that contain PII:

> 1. Annotated PII datasets: These datasets are specifically designed to train PII Detection Models, which are used to detect and mask PII. For example, these models can help with online content moderation or provide anonymized databases.

Reviewer 3: Would be cool to have some examples here!
Are these what you're referring to (from a search for "PII detection")?
https://huggingface.co/datasets/melaniasala/PII_Data_Detection
https://huggingface.co/datasets/metaboulie/Tidied-PII-Detection-Kaggle-7k

Author: Yes exactly ! I often mention https://huggingface.co/datasets/ai4privacy/pii-masking-300k since we enabled the Presidio detection widget on this one and us well documented. I'll add this one as exemple :)

## A New Experiment on the Dataset Hub: Presidio Reports

To help address these challenges, we're experimenting with a new feature on the Dataset Hub that uses [Presidio](https://github.com/microsoft/presidio), an open-source state-of-the-art PII detection tool.
Presidio relies on detection patterns and machine learning models to identify PII.

With this new feature, users will be able to see a report that estimates the presence of PII in a dataset.
This information can be valuable for ML practitioners, helping them make informed decisions before training a model.
For example, if the report indicates that a dataset contains sensitive PII, practitioners may choose to further filter the dataset using tools like Presidio.

Dataset owners can also benefit from this feature by using the reports to validate their PII filtering processes before releasing a dataset.

## An Example of a Presidio Report

Let's take a look at an example of a Presidio report for this [pre-training dataset](https://huggingface.co/datasets/allenai/c4):

> ![Presidio report](assets/presidio-pii-detection/presidio_report.jpg)

Reviewer 4: ```suggestion
![Presidio report](assets/presidio-pii-detection/presidio_report.png)
```
